ledger/participant-integration-api/src/test/lib/scala/platform/indexer/ha/EndlessReadService.scala

```diff
@@ -21,18 +20,18 @@ import com.daml.lf.value.Value
 import com.daml.logging.{ContextualizedLogger, LoggingContext}
 import com.google.protobuf.ByteString
 
-import java.util.concurrent.atomic.AtomicInteger
-import scala.concurrent.duration._
+import java.time.Instant
 
 /** An infinite stream of state updates that fully conforms to the Daml ledger model.
   *
   *  All instances of this class produce the same stream of state updates.
   *
-  *  @param updatesPerSecond The maximum number of updates per second produced.
+  *  @param additionalSourceFlow Additional stream Flow after the offset to extend the service with additional
+  *                              functionality like throttling,  etc.
   */
 case class EndlessReadService(
-    updatesPerSecond: Int,
     name: String,
+    additionalSourceFlow: Flow[Int, Int, NotUsed] = Flow[Int],
```

Comment:
As discussed online, you don't need to remove the throttling for the purpose of the stub indexer benchmark. I think I would also prefer a `class ThrottledReadService(delegate: ReadService)` wrapper that throttles a given read service, instead of parametrizing `EndlessReadService`.

Reply:
ok, this can be done as separate PR